cli/flox-rust-sdk/src/models/manifest/composite.rs

```diff
@@ -0,0 +1,276 @@
+        merged.extend(manifest2.vars.iter().map(|(k, v)| (k.clone(), v.clone())));
+        Ok(merged)
+    }
+
+    /// TODO: Not implemented.
+    fn merge_hook(
+        _manifest1: &Manifest,
+        _manifest2: &Manifest,
+    ) -> Result<ManifestHook, MergeError> {
+        Ok(ManifestHook::default())
+    }
+
+    /// TODO: Not implemented.
+    fn merge_profile(
+        _manifest1: &Manifest,
+        _manifest2: &Manifest,
+    ) -> Result<ManifestProfile, MergeError> {
+        Ok(ManifestProfile::default())
+    }
+
+    /// TODO: Not implemented.
+    fn merge_options(
+        _manifest1: &Manifest,
+        _manifest2: &Manifest,
+    ) -> Result<ManifestOptions, MergeError> {
+        Ok(ManifestOptions::default())
+    }
+
+    /// TODO: Not implemented.
+    fn merge_services(
+        _manifest1: &Manifest,
+        _manifest2: &Manifest,
+    ) -> Result<ManifestServices, MergeError> {
+        Ok(ManifestServices::default())
+    }
+
+    /// TODO: Not implemented.
+    fn merge_build(
+        _manifest1: &Manifest,
+        _manifest2: &Manifest,
+    ) -> Result<ManifestBuild, MergeError> {
+        Ok(ManifestBuild::default())
+    }
+
+    /// TODO: Not implemented.
+    fn merge_containerize(
+        _manifest1: &Manifest,
+        _manifest2: &Manifest,
+    ) -> Result<Option<ManifestContainerize>, MergeError> {
+        Ok(None)
+    }
+
+    fn merge(&self, manifest1: Manifest, manifest2: Manifest) -> Result<Manifest, MergeError> {
+        let manifest = Manifest {
+            version: Self::merge_version(&manifest1, &manifest2)?,
+            install: Self::merge_install(&manifest1, &manifest2)?,
+            vars: Self::merge_vars(&manifest1, &manifest2)?,
+            hook: Self::merge_hook(&manifest1, &manifest2)?,
+            profile: Self::merge_profile(&manifest1, &manifest2)?,
+            options: Self::merge_options(&manifest1, &manifest2)?,
+            services: Self::merge_services(&manifest1, &manifest2)?,
+            build: Self::merge_build(&manifest1, &manifest2)?,
+            containerize: Self::merge_containerize(&manifest1, &manifest2)?,
+        };
+
+        Ok(manifest)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::BTreeMap;
+
+    use pretty_assertions::assert_eq;
+
+    use super::*;
+
+    #[test]
+    fn shallow_merger_no_deps() {
+        let composer = Manifest {
+            version: Version::<1>,
+            vars: ManifestVariables(BTreeMap::from([(
+                "composer-keep".to_string(),
+                "composer".to_string(),
+            )])),
+            ..Manifest::default()
+        };
+
+        let composite_manifest = CompositeManifest {
+            composer,
+            deps: vec![],
+        };
+
+        let merged = composite_manifest.merge(ShallowMerger).unwrap();
+
+        assert_eq!(merged, Manifest {
+            version: Version::<1>,
+            vars: ManifestVariables(BTreeMap::from([(
```

Comment:
You may be able to avoid writing all of this out by using the "struct update syntax" e.g.
```
Manifest {
    version: Version::<1>,
    vars: ...,
    ..Default::default()
}
```